procdesc.cc

```diff
@@ -245,19 +249,36 @@ static void *SubThreadMain(void *) {
 
 static void *ThreadMain(void *) {
   int pd;
+  int pipefds[2];
+  EXPECT_EQ(0, pipe(pipefds));
   pid_t child = pdfork(&pd, 0);
   if (child == 0) {
+    close(pipefds[0]);
     // Child: start a subthread then loop
     pthread_t child_subthread;
-    EXPECT_OK(pthread_create(&child_subthread, NULL, SubThreadMain, NULL));
+    // Wait for the subthread startup using atomics and p.
+    // Note: using the __sync compiler intrinsics to avoid depending on C++11.
```

Comment:
Why does this need to use a new and different synchronization primitive?  Wouldn't it just be possible to create another pipe and pass one of the file descriptors as an argument to the new thread, then sync with `SEND_INT_MESSAGE` ?

Reply:
Yes good point, not sure why I thought I shouldn't be using pipes here.